lib/Genesis/Legacy.pm

```diff
@@ -280,11 +280,19 @@ sub cert_commands {
 	my $missing_only = ($options{scope}||'') eq 'add';
 	for my $path (sort keys %$certs) {
 		my $rand = sprintf("%09d", rand(1000000000));
+		my $ttl = "1y";
+		my @names = ('--name', "ca.n$rand.$path");
+		if (defined($certs->{$path}{"ca"})) {
+			$ttl = $certs->{$path}{"ca"}{valid_for} || $ttl;
+			push(@names, '--name', $_)
+				for (@{ $certs->{$path}{"ca"}{names} || [] });
```

Comment:
This doesn't allow users to override the CN of the CA certificate, just the SANs.

Reply:
This was due to the discussion that we don't want users supplying potentially breaking names, but we still want to support them being able to supply names in some meaningful fashion.

Reply:
I think we need to allow users to supply breaking names, if they want to.  We just need to randomize if missing, and then never set the CA name in a kit.